arangod/Replication/DatabaseInitialSyncer.cpp

```diff
@@ -1307,6 +1307,10 @@ Result DatabaseInitialSyncer::fetchCollectionSyncByRevisions(arangodb::LogicalCo
   //std::pair<std::size_t, std::size_t> fullRange = treeLeader->range();
   std::unique_ptr<containers::RevisionTree> treeLocal =
       physical->revisionTree(*trx);
+  if (!treeLocal) {
+    // local collection does not support syncing by revision, fall back to keys
+    return fetchCollectionSyncByKeys(coll, leaderColl, maxTick);
```

Comment:
This will make `fetchCollectionSyncByKeys()` run inside the transaction started in line 1293.
Do we need to abort this transaction via `guard.fire();` before we branch into `fetchCollectionSyncByKeys()` here?

Reply:
Good catch.